CMakeRC.cmake

```diff
@@ -143,15 +146,16 @@ struct created_subdirectory {
 };
 
 class directory {
-    std::deque<file_data> _files;
-    std::deque<directory> _dirs;
+    std::list<file_data> _files;
+    std::list<directory> _dirs;
     std::map<std::string, file_or_directory> _index;
 
     using base_iterator = std::map<std::string, file_or_directory>::const_iterator;
 
 public:
 
     directory() = default;
+    directory(directory&&) = default;
```

Comment:
This shouldn't be required if you are using `list`. The immovability of this class is required since the `embedded_filesystem` maintains a tangled mess of pointers between things.

Reply:
Yeah, stepped in that mess when I used a std::vector on my first pass at this!